src/draft/debugDebug.ts

```diff
@@ -0,0 +1,55 @@
+import * as vscode from 'vscode';
+import { LoggingDebugSession, InitializedEvent, TerminatedEvent } from 'vscode-debugadapter';
+import { EventEmitter } from 'events';
+import { ChildProcess, exec } from 'child_process';
+import { OutputChannel } from 'vscode';
+import { Readable } from 'stream';
+import { DraftRuntime } from './draftRuntime';
+import { DebugProtocol } from 'vscode-debugprotocol/lib/debugProtocol';
+const { Subject } = require('await-notify');
+
+
+export class DraftDebugSession extends LoggingDebugSession {
```

Comment:
From the implementation, every F5 will create two debug sessions, one is draft, another is the real logic application debug session. Feel a little weird. Is the draft debug session necessary?

Reply:
`DraftConfigurationProvider#resolveDebugConfiguration` is actually the recommended way to intercept F5. Inside the function, you can run `draft` command to prepare the environment, and then invoke startDebugging API to launch third-party debug session like node/java/go, but let resolveDebugConfiguration function itself return `undefined`. If its return value is undefined, VS Code will not launch the second debug session.

The second way is to take advantage of `preLaunchTask`. It's designed to run build and deploy before launching debug session.

Reply:
@testforstephen For now I think we need the Draft session to support the file watcher functionality. For the next iteration I'm looking into not having to press the stop debug two times.